src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2381,6 +2381,11 @@ internal Binder GetBinder(CSharpSyntaxNode syntaxNode)
                     {
                         diagnostics.Add(ErrorCode.ERR_PartialMethodMustHaveLatent, method.Locations[0], method);
                     }
+
+                    if (method.IsPartialDefinition && (object)method.PartialImplementationPart == null)
+                    {
+                        method.DeclaringCompilation.SymbolDeclaredEvent(method);
```

Comment:
I don't understand what you are suggesting. Which condition, how are you suggesting it be adjusted, and instead of what?

Events for partial methods have to be delayed so that the symbol is not exposed until the partial definition and the partial declaration have been paired up.